src/app/devices/deviceList/components/deviceList.tsx

```diff
@@ -118,132 +132,130 @@ class DeviceListComponent extends React.Component<DeviceListDataProps & DeviceLi
     }
 
     private readonly showDeviceList = (context: LocalizationContextInterface) => {
-
-        const renderCell = (nestingDepth: number, item: DeviceSummary, itemIndex: number) => {
-            return (
-                <DeviceListCell
-                    connectionString={this.props.connectionString}
-                    device={item}
-                    itemIndex={itemIndex}
-                />
-            );
-        };
-
         return (
             <>
+                {this.showPaging()}
+                <div className="list-detail">
+                    {this.props.isFetching ?
+                        <MultiLineShimmer shimmerCount={SHIMMER_COUNT}/> :
+                        (this.props.devices && this.props.devices.length !== 0 ?
+                            <MarqueeSelection selection={this.selection}>
+                                <DetailsList
+                                    onRenderItemColumn={this.renderItemColumn(context)}
+                                    items={!this.props.isFetching && this.props.devices}
+                                    columns={this.getColumns(context)}
+                                    layoutMode={DetailsListLayoutMode.justified}
+                                    selection={this.selection}
+                                />
+                            </MarqueeSelection> :
+                            <>
+                                <h3>{context.t(ResourceKeys.deviceLists.noDevice)}</h3>
+                                <Announced
+                                    message={context.t(ResourceKeys.deviceLists.noDevice)}
+                                />
+                            </>
+                        )
+                    }
+                </div>
+            </>
+        );
+        // todo announce no result
```

Comment:
Isn't this todo covered in lines 153-155?

Reply:
Yeah. I forgot to remove the comment afterwards. Fixing.